libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Tests.hs

```diff
@@ -699,7 +700,7 @@ runPreds ps = do
 -- | Generate a set of satisfiable constraints and check that we can generate a solution and that it
 --   actually satisfies the constraints.
 prop_soundness :: OrderInfo -> Property
-prop_soundness = prop_soundness' False []
+prop_soundness x = within 2_000_000 $ prop_soundness' False [] x
```

Comment:
I think it needs to be adjusted in the `prop_soundness'`, because it is an argument to:
https://github.com/IntersectMBO/cardano-ledger/blob/d3e92cf43ab9408180944604fed5210534afdd2b/libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Tests.hs#L653C20-L653C27

Reply:
So a little bit of research shows there are two ways we might handle this

1) 
within :: [Testable](https://hackage.haskell.org/package/QuickCheck-2.14.3/docs/Test-QuickCheck.html#t:Testable) prop => [Int](https://hackage.haskell.org/package/base-4.16.3.0/docs/Data-Int.html#t:Int) -> prop -> [Property](https://hackage.haskell.org/package/QuickCheck-2.14.3/docs/Test-QuickCheck.html#t:Property)

Considers a property failed if it does not complete [within](https://hackage.haskell.org/package/QuickCheck-2.14.3/docs/Test-QuickCheck.html#v:within) the given number of microseconds.

Note: if the property times out, variables quantified inside the [within](https://hackage.haskell.org/package/QuickCheck-2.14.3/docs/Test-QuickCheck.html#v:within) will not be printed. Therefore, you should use within only in the body of your property.

Good: prop_foo a b c = within 1000000 ...

Bad: prop_foo = within 1000000 $ \a b c -> ...

Bad: prop_foo a b c = ...; main = quickCheck (within 1000000 prop_foo)

So it seems that the inner most within controls things. So the suggested soulution would be the inner one, So that should work.

2)
discardAfter :: [Testable](https://hackage.haskell.org/package/QuickCheck-2.14.3/docs/Test-QuickCheck.html#t:Testable) prop => [Int](https://hackage.haskell.org/package/base-4.16.3.0/docs/Data-Int.html#t:Int) -> prop -> [Property](https://hackage.haskell.org/package/QuickCheck-2.14.3/docs/Test-QuickCheck.html#t:Property)

Discards the test case if it does not complete within the given number of microseconds. This can be useful when testing algorithms that have pathological cases where they run extremely slowly.

So if we add discard after, then the test is discarded. That might be the best solution, since this test succeeds quite quickly in most cases. So if we generate  a pathalogical case, we just discard.

Your call.

Reply:
We should fail, instead of discard. 
First of all `discardAfter` could mask a bug that occasionally triggers an infinite loop. Second of all, I always considered discards a hack that people use when they can't figure out how to fix their test properly.

Moreover, even if there is no infinite loop, the pathological case should not take too long to run. 

Note that current value is set to `0.1` second, not `1` second. So increasing it to `2` should give it plenty of time to terminate.

Reply:
So following Max's advice, we will just double the amount passed to constraintProperty , that way it still protects against infinite loops, and gives extra lee way.